cjwkernel/pandas/types.py

```diff
@@ -411,12 +411,19 @@ class I18nMessage:
     args: Dict[str, Union[int, float, str]] = field(default_factory=dict)
     """Arguments (empty if message does not need any -- which is common)."""
 
+    source: Dict[str, str] = field(default_factory=dict)
```

Comment:
I think this should be `Optional[Dict[...]] = None`, to be consistent with my suggestions in `moduletypes.py` and `cjwkernel/types.py`.